request_handlers_test.cc

```diff
@@ -0,0 +1,111 @@
+#include "request_handlers.h"
+#include "gtest/gtest.h"
+
+#include <iostream>
+
+using namespace lightning_server;
+
+const int MAX_REQ_SIZE = 8192; // bytes
+
+class RequestHandlersTest : public ::testing::Test {
+protected:
+  // Helper method to pass arbitrary strings into the parser
+  bool HandleRequest(char* request_buffer,
+                     char* response_buffer,
+                     size_t& response_buffer_size) {
+    request_handlers::echoRequestHandler(request_buffer,
+                                         response_buffer,
+                                         response_buffer_size);
+    if (response_buffer == nullptr) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+};
+
+TEST_F(RequestHandlersTest, EmptyStringTest) {
+  char request_buffer[MAX_REQ_SIZE];
+
+  // Customize message in request buffer
+  strcpy(request_buffer, "");
+
+  // Handle echo response in external handler
+  char response_buffer[MAX_REQ_SIZE];
+  size_t response_size = 0;
+
+  EXPECT_TRUE(HandleRequest(request_buffer, response_buffer, response_size))
```

Comment:
Could this still pass if HandleRequest set response_buffer to anything that isn't nullptr, like `'sadl;fkjakd'`?